### PR TITLE
fix(checkout): fix checkout step redirection logic for logged-in users & allow editing of shipping address & delivery method

### PR DESCRIPTION
**Pull request by gajda-w**
I want to merge this change because it fixes the checkout flow by ensuring that logged-in users can properly navigate through the checkout steps. This update allows users to stay on the "Shipping Address" and "Delivery Method" steps to edit details if needed, and ensures they are redirected to the appropriate step only when necessary (e.g. missing email, shipping address, or delivery method).

# Pull Request Checklist

- [ ] Test the changes locally to ensure they work as expected.
- [ ] Document the testing process and results in the pull request description. (Screen recording, screenshot etc)
- [ ] Include new tests for any new functionality or significant changes.
- [ ] Ensure that tests cover edge cases and potential failure points.
- [ ] Document the impact of the changes on the system, including potential risks and benefits.
- [ ] Provide a rollback plan in case the changes introduce critical issues.
- [ ] Update documentation to reflect any changes in functionality.

### CODE REVIEW

File: apps/storefront/src/app/[locale]/(checkout)/checkout/actions.ts [hunk 1]
```diff
@@ -30,14 +32,23 @@ export const validateCheckoutStepAction = async ({
...
+  // If missing, redirect to shipping address step
+  if (!checkout.shippingAddress) {
     if (step !== "shipping-address") {
       redirect({ href: paths.checkout.shippingAddress.asPath(), locale });
     }
 
     return;
   }
 
+  // Allow user to stay on shipping-address step (e.g. to edit it)
```
Comment (soniaklimas) [inline apps/storefront/src/app/[locale]/(checkout)/checkout/actions.ts:45]:
```suggestion
  // Allow user to stay on shipping address step (e.g. to edit it)
```

**Review by soniaklimas — APPROVED** [review]
